### ISSUE TITLE
[Feature Request] Smaller Int Types

### ISSUE THREAD

**Issue by DalenW** [typesense/typesense#878]
Sorry if this is the wrong way to request this, but I would like smaller int field types. 

A lot of times I index enumerators, which rarely go above int8 or int16. I'd save a bit of memory with smaller int types.

**Comment by kishorenc** [comment]
I agree, it would be good to conserve memory with smaller types. int8 and int16 will be useful.

**Comment by DalenW** [comment]
Thank you!
